app/services/order_cycle_distributed_products.rb

```diff
@@ -9,7 +9,7 @@ def initialize(distributor, order_cycle, customer)
   end
 
   def products_relation
-    Spree::Product.where(id: stocked_products)
+    Spree::Product.where(id: stocked_products).group("spree_products.id")
```

Comment:
Is this is an unusual use of `group`? Why didn't you go for the same approach as before? `select("spree_products.*")`

Reply:
I wanted to make sure there were no duplicates in the results, and initially tried using DISTINCT in the query here, but I ran into an obscure Rails/Postgres issue when the taxon ordering feature is applied. It caused a lot of fatal errors and took a while to figure out. I settled on using group by id as a workaround.

Further reading: 

https://stackoverflow.com/questions/12693089/pgerror-select-distinct-order-by-expressions-must-appear-in-select-list/18367248

https://github.com/rails/rails/issues/520

Reply:
Ah, okay, I ran into that issue before. It's frustrating.